src/dlgProfilePreferences.cpp

```diff
@@ -1918,8 +1918,8 @@ void dlgProfilePreferences::copyMap()
     QSetIterator<Host*> itActiveHost(activeHosts);
     while (itActiveHost.hasNext()) {
         Host* pOtherHost = itActiveHost.next();
-        if (pOtherHost && pHost != pOtherHost && pOtherHost) {
-            QString otherHostName = pOtherHost->getName();
+        if (pOtherHost != pHost) {
```

Comment:
From what I remember of this only the last `&& pOtherHost` is superfluous. This should be:
```
    if (pOtherHost && (pHost != pOtherHost)) {
```
Not sure if that inner pair of parentheses is actually needed but perhaps it helps to clarify the intent...

Reply:
Are you certain `pOtherHost` can be null - `itActiveHost.hasNext()` right before?

Reply:
Just because it returns a value there is no reason why it has to be valid!? How do we guarantee something else had not just closed the profile and then the underlying `QPointer<Host>` (?) will drop back to `nullptr` won't it? OTOH if destroying profile Host instances is so bloody unlikely to occur then why do we always check for non-null `Host` pointers everywhere? :see_no_evil:

Reply:
No clue